packages/loadable-adapters/src/federation-stats-plugin/federation-stats-plugin.js

```diff
@@ -91,6 +95,45 @@ export class FederationStatsPlugin {
           }
         }
       );
+
+      const hasSriPlugin = Boolean(
+        compiler.options.plugins &&
+          compiler.options.plugins.find((plugin) => plugin.constructor.name === 'SubresourceIntegrityPlugin')
+      );
+
+      if (hasSriPlugin) {
+        // needs to be optimized
+        const assetIntegrityMap = new Map();
+
+        compiler.hooks.done.tap(PLUGIN_NAME, (stats) => {
+          assetIntegrityMap.clear();
+
+          stats
+            .toJson()
+            .assets.filter((asset) => Boolean(asset.integrity))
+            .forEach((asset) => {
+              const integrity = Array.isArray(asset.integrity) ? asset.integrity[0] : asset.integrity;
+              console.log(asset.name, integrity);
+              assetIntegrityMap.set(asset.name, integrity);
+            });
+        });
+
+        compiler.hooks.afterDone.tap(PLUGIN_NAME, () => {
+          const fileName = this._options.fileName;
+          const statsFilePath = path.join(compiler.options.output.path, fileName);
+          const rawMfStats = fs.readFileSync(statsFilePath, 'utf-8');
```

Comment:
file exists check and proper error?
and wrap json.parse in try...catch?

Reply:
Solved :+1: 
https://github.com/valorkin/mf-loadable-adapter/pull/2/commits/adbb6f322477788f3b37cb56668ca133cebc36c5